lib/ansible/modules/system/etc_hosts.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python
+# encoding: utf-8
+
+# (c) 2016, Jiri Tyr <jiri.tyr@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils._text import to_bytes
+import os
+import re
+import tempfile
+
+
+ANSIBLE_METADATA = {
+    'status': [
+        'preview'
+    ],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: etc_hosts
+author: Jiri Tyr (@jtyr)
+version_added: '2.3'
+short_description: Manages records in the /etc/hosts file
+description:
+  - Add, remove and modify records in the /etc/hosts file.
+options:
+  ip:
+    required: true
+    description:
+      - IP address.
+  hostname:
+    required: false
+    description:
+      - Canonical hostname of the IP address. Required if I(state=present).
+  alias:
+    required: false
+    default: null
+    description:
+      - Alias(es) of the IP address. Multiple aliases can be either a normal
+        YAML list or a string containing comma separated list.
+  path:
+    required: false
+    default: /etc/hosts
```

Comment:
Wouldn't it be better to adjust the name of the module to `hosts` ? It seems odd to effectively hardcode the filepath in the module name, while at the same time allowing to adjust the file path. This module might also be extended to add support for managing the Windows  `hosts` file,

Furthermore both Puppet and Salt call their analogue type/modules also `hosts`.